lib/registries/addon/overview/route.ts

```diff
@@ -76,23 +79,68 @@ export default class Overview extends GuidRoute {
                 ),
                 institution: (institutions as SparseModel[]).map(institution => institution.name as string),
             };
-            const headTags = [...this.metaTags.getHeadTags(metaTagsData)];
+
+            const metaTags: HeadTagDef[] = this.metaTags.getHeadTags(metaTagsData);
+
+            // Google Structured Data
+            let jsonLD: object | undefined;
+            let scriptTag;
+            try {
+                jsonLD = await this.returnStructuredData(id);
+                const jsonString: object | undefined = jsonLD ? jsonLD : undefined;
+                if (jsonString) {
+                    scriptTag = {
+                        type: 'script',
+                        content: jsonString,
+                        attrs: {
+                            type: 'application/ld+json',
+                        },
+                    };
+                } else {
+                    scriptTag = undefined;
+                }
+                if (scriptTag) {
+                    metaTags.push(scriptTag);
+                }
+            } catch (e) {
+                const errorMessage = this.intl.t('general.structured_data.json_ld_retrieval_error');
+                captureException(e, { errorMessage });
+            }
+
             if (provider && provider.assets && provider.assets.favicon) {
-                headTags.push({
+                metaTags.push({
                     type: 'link',
                     attrs: {
                         rel: 'icon',
                         href: provider.assets.favicon,
                     },
                 });
             }
-            this.set('headTags', headTags);
+            this.set('headTags', metaTags);
             this.metaTags.updateHeadTags();
         }
-
         blocker.done();
     }
 
+    async returnStructuredData(guid: string): Promise<object | undefined> {
+        const path = `${config.OSF.url}/${guid}/metadata/?format=google-dataset-json-ld`;
+        let jsonLD;
+        let jsonFetch: object | undefined;
+        try {
+            jsonFetch = await this.currentUser.authenticatedAJAX({
+                method: 'GET',
+                url: path,
+            });
+            if (jsonFetch) {
+                jsonLD = jsonFetch;
+            }
+        } catch (e) {
+            const errorMessage = this.intl.t('general.structured_data.json_ld_retrieval_error');
+            captureException(e, { errorMessage });
+        }
+        return jsonLD;
+    }
+
```

Comment:
```suggestion
    async returnStructuredData(guid: string): Promise<object | undefined> {
        const path = `${config.OSF.url}/${guid}/metadata/?format=google-dataset-json-ld`;
        return this.currentUser.authenticatedAJAX({
            method: 'GET',
            url: path,
        });
    }

```
We can either handle the exception here, or in `setHeadTags`, but not both, since you are not re-throwing the exception. Simply returning the promise here is sufficient.